hw1/02-palindrome.js

```diff
@@ -1,3 +1,43 @@
 const elem = document.querySelector('input');
-
 elem.addEventListener('input', handleInput);
+const output = document.querySelector('.output');
+
+function handleInput() {
+  const user_input = elem.value;
+  return palindrome_check(user_input);
+}
+
+function palindrome_check(user_input) {
+  if (user_input < 0) {
+    return input_err();
+  } else if (user_input === '') {
+    return input_err();
+  } else if (
+    user_input.toString() === user_input.toString().split('').reverse().join('')
+  ) {
+    return pali();
+  } else {
+    return non_pali();
+  }
+}
+
+function non_pali() {
+  output.style.color = 'red';
+  output.innerHTML = 'No. Try Again';
+  console.log('False');
+  return;
+}
+
+function pali() {
+  output.style.color = 'green';
+  output.innerHTML = 'Yes. This is a palindrome!';
+  console.log('True');
+  return;
+}
+
+function input_err() {
+  output.style.color = 'red';
+  output.innerHTML = 'Error. Please enter a positive number.';
+  console.log('False');
+  return false;
```

Comment:
I'm not sure if this function should really return a boolean value.  From the looks of it the function call doesn't really catch the value either.  This function always returns false no matter what so maybe we just change it to `return` instead.